File: src/test/java/ru/ezhov/rocket/action/infrastructure/YmlRocketActionSettingsRepositoryTest.java
package ru.ezhov.rocket.action.infrastructure;

import org.junit.Ignore;
import org.junit.Test;
import ru.ezhov.rocket.action.RocketActionSettingsRepositoryException;
import ru.ezhov.rocket.action.api.RocketActionSettings;

import java.io.File;
import java.net.URISyntaxException;
import java.util.Arrays;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static org.junit.Assert.assertEquals;

public class YmlRocketActionSettingsRepositoryTest {

    @Test
    public void actions() throws URISyntaxException, RocketActionSettingsRepositoryException {
        YmlRocketActionSettingsRepository repository =
                new YmlRocketActionSettingsRepository(
                        this.getClass().getResource("/actions.yml").toURI()
                );

        final List<RocketActionSettings> actions = repository.actions();

        assertEquals(18, actions.size());

        System.out.println(actions);
    }

    @Test
    @Ignore
    public void save() throws URISyntaxException, RocketActionSettingsRepositoryException {
        YmlRocketActionSettingsRepository repository =
                new YmlRocketActionSettingsRepository(
                        new File("./test.yml").toURI()
                );

        repository.save(Arrays.asList(
                new RocketActionSettings() {
                    @Override
                    public String type() {
                        return "test";
                    }

                    @Override
                    public Map<String, String> settings() {
                        Map<String, String> map = new HashMap<>();
                        map.put("1", "2");
                        return map;
                    }

                    @Override
                    public List<RocketActionSettings> actions() {
                        return Collections.emptyList();
                    }
                },
                new RocketActionSettings() {
                    @Override
                    public String type() {
                        return "test";
                    }

                    @Override
                    public Map<String, String> settings() {
                        Map<String, String> map = new HashMap<>();
                        map.put("1", "2");
                        return map;
                    }

                    @Override
                    public List<RocketActionSettings> actions() {
                        return Arrays.asList(
                                new RocketActionSettings() {
                                    @Override
                                    public String type() {
                                        return "test";
                                    }

                                    @Override
                                    public Map<String, String> settings() {
                                        Map<String, String> map = new HashMap<>();
                                        map.put("1", "2");
                                        return map;
                                    }

                                    @Override
                                    public List<RocketActionSettings> actions() {
                                        return Collections.emptyList();
                                    }
                                },
                                new RocketActionSettings() {
                                    @Override
                                    public String type() {
                                        return "test";
                                    }

                                    @Override
                                    public Map<String, String> settings() {
                                        Map<String, String> map = new HashMap<>();
                                        map.put("1", "2");
                                        return map;
                                    }

                                    @Override
                                    public List<RocketActionSettings> actions() {
                                        return Collections.emptyList();
                                    }
                                }
                        );
                    }
                }
        ));
    }
}

File: src/test/kotlin/ru/ezhov/rocket/action/infrastructure/YmlRocketActionSettingsRepositoryTest.kt
package ru.ezhov.rocket.action.infrastructure

import org.junit.Assert
import org.junit.Ignore
import org.junit.Test
import ru.ezhov.rocket.action.domain.RocketActionSettingsRepositoryException
import ru.ezhov.rocket.action.api.RocketActionSettings
import java.io.File
import java.net.URISyntaxException
import java.util.*

class YmlRocketActionSettingsRepositoryTest {
    @Test
    @Throws(URISyntaxException::class, RocketActionSettingsRepositoryException::class)
    fun actions() {
        val repository = YmlRocketActionSettingsRepository(
                this.javaClass.getResource("/actions.yml")!!.toURI()
        )
        val actions = repository.actions()
        Assert.assertEquals(20, actions.size.toLong())
        println(actions)
    }

    @Test
    @Ignore
    @Throws(URISyntaxException::class, RocketActionSettingsRepositoryException::class)
    fun save() {
        val repository = YmlRocketActionSettingsRepository(
                File("./test.yml").toURI()
        )
        repository.save(listOf(
                object : RocketActionSettings {
                    override fun id(): String {
                        return "id1"
                    }

                    override fun type(): String {
                        return "test"
                    }

                    override fun settings(): Map<String, String> {
                        val map: MutableMap<String, String> = HashMap()
                        map["1"] = "2"
                        return map
                    }

                    override fun actions(): List<RocketActionSettings> {
                        return emptyList()
                    }
                },
                object : RocketActionSettings {
                    override fun id(): String {
                        return "id2"
                    }

                    override fun type(): String {
                        return "test"
                    }

                    override fun settings(): Map<String, String> {
                        val map: MutableMap<String, String> = HashMap()
                        map["1"] = "2"
                        return map
                    }

                    override fun actions(): List<RocketActionSettings> {
                        return listOf(
                                object : RocketActionSettings {
                                    override fun id(): String {
                                        return "id4"
                                    }

                                    override fun type(): String {
                                        return "test"
                                    }

                                    override fun settings(): Map<String, String> {
                                        val map: MutableMap<String, String> = HashMap()
                                        map["1"] = "2"
                                        return map
                                    }

                                    override fun actions(): List<RocketActionSettings> {
                                        return emptyList()
                                    }
                                },
                                object : RocketActionSettings {
                                    override fun id(): String {
                                        return "id5"
                                    }

                                    override fun type(): String {
                                        return "test"
                                    }

                                    override fun settings(): Map<String, String> {
                                        val map: MutableMap<String, String> = HashMap()
                                        map["1"] = "2"
                                        return map
                                    }

                                    override fun actions(): List<RocketActionSettings> {
                                        return emptyList()
                                    }
                                }
                        )
                    }
                }
        ))
    }
}